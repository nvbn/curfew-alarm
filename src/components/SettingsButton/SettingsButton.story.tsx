import { storiesOf } from "@storybook/react-native";
import React from "react";

import SettingsButton from "./SettingsButton";

storiesOf("<SettingsButton />", module).add("default", () => (
  <SettingsButton onPress={() => alert("pressed!")} />
));
