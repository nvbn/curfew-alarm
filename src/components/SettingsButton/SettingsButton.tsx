import { Ionicons } from "@expo/vector-icons";
import React from "react";
import { TouchableOpacity } from "react-native";

import styles from "./styles";

type Props = {
  onPress: () => void;
};

const SettingsButton = ({ onPress }: Props): JSX.Element => (
  <TouchableOpacity onPress={onPress} style={styles.button}>
    <Ionicons name="ios-settings-outline" size={24} color="black" />
  </TouchableOpacity>
);

export default SettingsButton;
