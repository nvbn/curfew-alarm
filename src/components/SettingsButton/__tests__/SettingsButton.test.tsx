import { render } from "@testing-library/react-native";
import React from "react";

import SettingsButton from "../SettingsButton";

describe("<SettingsButton />", () => {
  test("can be rendered", () => {
    const rendered = render(<SettingsButton onPress={jest.fn()} />);

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
