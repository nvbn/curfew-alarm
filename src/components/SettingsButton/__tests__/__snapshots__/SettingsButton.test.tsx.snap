// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsButton /> can be rendered 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingLeft": 10,
      "paddingRight": 10,
    }
  }
>
  <Text />
</View>
`;
